### DISCUSSION TITLE
请问，我参考use-antd项目实现自定义编辑器，如何在tsx里读取到打开的文件路径以及内容？

### DISCUSSION THREAD

**Discussion by hnhhzy** [opensumi/core#2352] [Q&A] [answered]
请问，我参考use-antd项目实现自定义编辑器，如何在tsx里读取到打开的文件路径以及内容？
其实就是use-antd的demo项目了。。。如何在tsx里读取到 index.antd-theme的内容

**Comment by hnhhzy** [comment]
https://github.com/opensumi/opensumi-module-samples/tree/main/modules/use-antd 这个项目的demo,在txs里如何读取到打开的这个文件地址以及内容？

**Reply by erha19** [reply]
没理解，这里打开的是一个自定义协议页面，没有具体文件，文件路径为 `antd-theme:\` , 你指的是点击 `index.antd-theme` 文件打开的页面吗？

**Reply by hnhhzy** [reply]
不是，假设 index.antd-theme 里面有内容，他是abcdefg，我想在components.view.tsx里 return ( <div>abcdefg</div> }
如何获取他原本文件的内容以及他的文件路径？

**Reply by erha19** [reply]
文件通过自定义组件打开，目前不支持直接传参，但你可以通过 `useInjectable(IFileServiceClient)` 方法，在组件内读取文件内容

**Reply by erha19** [reply]
同时，在使用 `registerEditorComponent` 注册自定义组件协议时，你也可以通过第二个参数，往渲染组件内传递初始 Props，可以通过这种手段将文件路径传递进去

**Reply by erha19** [reply]
案例内，渲染 AntdComponent 时选择了 `renderMode: EditorComponentRenderMode.ONE_PER_WORKBENCH` 参数，即全局会复用同一个组件，组件只会被初始化一次，如果希望对同个组件做多实例（动态）创建，可以去掉这个参数，采用默认的组件创建策略即可。

**Reply by hnhhzy** [reply]
我有看到registerEditorComponent的第二个参数 initialProps，但是传了以后，如何在tsx里面接收这个参数呢。

**Reply by hnhhzy** [reply]
useInjectable(IFileServiceClient)  这个组内读取我知道，在什么接口获取到我打开的这个文件的路径呢？

**Reply by erha19** [reply]
@hnhhzy 看一下这个评论 https://github.com/opensumi/core/discussions/2352#discussioncomment-5164767 ，你去掉这个参数应该就可以在你的组件里面获取到你传的属性的。

**Comment by hnhhzy** [comment, marked as answer]
const test = useInjectable<WorkbenchEditorService>(WorkbenchEditorService);
const data = test.currentResource?.uri;
好像可以了……

**Comment by life2015** [comment]
自定义编辑器component时，可以看下props里面有个一个resource变量，在这里你可以拿到想要的内容。
同时在 service.registerResourceProvider 方法中，可以传入自己想要的resource。

后续我会在samples补充一下这块的用法。
